src/core/net/srp_client.cpp

```diff
@@ -434,11 +435,18 @@ void Client::HandleRoleChanged(void)
 {
     if (Get<Mle::Mle>().IsAttached())
     {
+        // Avoid using long delays for initial registration
+        if (false == mIsThreadRoleAttached)
+        {
+            mUseUpdateTxDelayLong = false;
+        }
+        mIsThreadRoleAttached = true;
```

Comment:
We can eliminate the `mIsThreadRoleAttached` boolean variable since the same is already tracked by the `State` variable, specifically the `kStatePaused` state.

We can simply move the `mUseUpdateTxDelayLong = true` assignment to after the `VerifyOrExit(GetState() == kStatePaused)` check, and it should function equivalently.